tests/Integration/Behaviour/Features/Context/Domain/Product/PackedProductsFeatureContext.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain\Product;
+
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\RemoveAllPackedProductsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Pack\Command\SetPackProductsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Pack\Query\GetPackedProductsDetails;
+use PrestaShop\PrestaShop\Core\Domain\Product\Pack\QueryResult\PackedProductDetails;
+
+class PackedProductsFeatureContext extends AbstractProductFeatureContext
```

Comment:
Be careful you are duplicating an existing context here as well:
https://github.com/PrestaShop/PrestaShop/blob/6ed7d62946cffa98edf14a5c064587202c9ac048/tests/Integration/Behaviour/Features/Context/Domain/Product/UpdatePackFeatureContext.php

Reply:
All the query and commands have been tested already So if you need to you can improve the existing feature if you need to:
https://github.com/PrestaShop/PrestaShop/blob/6cabd15346f84ab6dec96b71c03ab495bebaad9e/tests/Integration/Behaviour/Features/Scenario/Product/update_pack.feature

But don't re-create everything or it will become a hell to maintain

Reply:
done